docs/design/coordinator.md

```diff
@@ -73,6 +73,13 @@ that are removed from the database.
 Segments that are overshadowed (their versions are too old and their data has been replaced by newer segments) are
 marked as unused. During the next Coordinator's run, they will be unloaded from Historical nodes in the cluster.
 
+### Cleaning up dangling tombstone segments
+
+On each run, the Druid coordinator will clean up any dangling tombstone segments.
+Dangling tombstones have eternity as their start or end -- for example,
```

Comment:
Thanks, updated javadocs to include some of this including some rationale!

>What about an infinity tombstone T that overlaps with an overshadowed segment A but has version older than A? (In this case, I would assume there is another segment B which is doing the actual overshadowing of A).
Here, would T still count as dangling? Or would it just get cleaned up by the regular MarkOvershadowedSegmentsAsUnused duty.

Correct, it would be the responsibility of `MarkOvershadowedSegmentsAsUnused` to mark an overshadowed tombstone. This duty only handles non-overshadowed eternity tombstones.